exporters/variables-scss/src/generators/__tests__/fileGenerator.test.ts

```diff
@@ -41,43 +97,163 @@ describe('fileGenerator', () => {
+          { fileName: 'other', content: mockedExpectedResult },
+        ],
+        expectedIndexFile: `@forward 'borders';\n@forward 'other';\n`,
+        description: 'should generate index file with multiple files',
+      },
+      {
+        files: [
+          { fileName: 'borders', content: emptyFile },
+          { fileName: 'other', content: mockedExpectedResult },
+        ],
+        expectedIndexFile: `export * from './borders';\nexport * from './other';\n`,
+        hasJsOutput: true,
+        description: 'should generate index file with one file with js output',
+      },
+    ];
+
+    it.each(dataProvider)('$description', ({ files, expectedIndexFile, hasJsOutput }) => {
+      const mockedIndexFile = generateIndexFile(files, hasJsOutput);
+
+      expect(mockedIndexFile).toBe(expectedIndexFile);
+    });
+  });
+
+  describe('generateThemesRootFile', () => {
+    it('should generate themes root file content', () => {
+      const themes = [{ name: 'theme-light' }, { name: 'theme-light-inverted' }];
+      const content = generateThemesRootFile(themes as TokenTheme[], false);
+
+      expect(content).toBe(mockedRootThemeFile);
+    });
+
+    it('should generate themes root file content with js output', () => {
+      const themes = [{ name: 'theme-light' }, { name: 'theme-light-inverted' }];
+      const content = generateThemesRootFile(themes as TokenTheme[], true);
+
+      expect(content).toBe(mockedRootThemeJsFile);
+    });
+  });
+
+  describe('generateRootThemesFileContent', () => {
+    it('should generate root themes file content', () => {
+      const themes = [{ name: 'theme-light' }, { name: 'theme-light-inverted' }];
+      const content = generateRootThemesFileContent(themes as TokenTheme[], false);
+
+      expect(content).toBe(
+        `theme-light: (\ncolors: theme-light.$colors,\n),\ntheme-light-inverted: (\ncolors: theme-light-inverted.$colors,\n),`,
+      );
+    });
+
+    it('should generate root themes file content with js output', () => {
+      const themes = [{ name: 'theme-light' }, { name: 'theme-light-inverted' }];
+      const content = generateRootThemesFileContent(themes as TokenTheme[], true);
+
+      expect(content).toBe(
+        `themeLight: {\ncolors: themeLight.colors\n},\nthemeLightInverted: {\ncolors: themeLightInverted.colors\n},`,
+      );
+    });
+  });
+
+  describe('generateRootThemesFileImports', () => {
+    it('should generate root themes file imports', () => {
+      const themes = [{ name: 'theme-light' }, { name: 'theme-light-inverted' }];
+      const content = generateRootThemesFileImports(themes as TokenTheme[], false);
+
+      expect(content).toBe(`@use 'themes/theme-light';\n@use 'themes/theme-light-inverted';`);
+    });
+
+    it('should generate root themes file imports with js output', () => {
+      const themes = [{ name: 'theme-light' }, { name: 'theme-light-inverted' }];
+      const content = generateRootThemesFileImports(themes as TokenTheme[], true);
```

Comment:
Just a note: if you want this test to be truly unit and independent, you should mock `prepareImportStatementCallback`. So the only thing you will be testing is a `map` and `join`. ;-)